python/lsst/daf/butler/formatters/parquet.py

```diff
@@ -1163,6 +1163,13 @@ def _numpy_style_arrays_to_arrow_arrays(
                 val = []
         else:
             val = np_style_arrays[name]
-        arrow_arrays.append(pa.array(val, type=schema.field(name).type))
+
+        try:
+            arrow_arrays.append(pa.array(val, type=schema.field(name).type))
+        except pa.ArrowNotImplementedError:
```

Comment:
This looks like it will send all `ArrowNotImplementedError` exceptions down to try  byte swap - is that what you want here? Or would some conditional checking work well, e.g.:
```
except pa.ArrowNotImplementedError as err:
    if "Byte-swapped arrays not supported" not in str(e):
        # try byte swapping
    else:
        # raise
```